subiquity/server/controllers/filesystem.py

```diff
@@ -351,29 +352,26 @@ async def v2_add_boot_partition_POST(self, disk_id: str) \
         self.add_boot_disk(disk)
         return await self.v2_GET()
 
-    async def v2_add_partition_POST(self, data: ModifyPartitionV2) \
+    async def v2_add_partition_POST(self, data: AddPartitionV2) \
             -> StorageResponseV2:
         log.debug(data)
         if data.partition.format is None:
             raise ValueError('add_partition must supply format')
         if data.partition.boot is not None:
             raise ValueError('add_partition does not support changing boot')
         disk = self.model._one(id=data.disk_id)
-        gap = gaps.largest_gap(disk)
-        if gap.size == 0:
```

Comment:
Huh it's not like it matters now but this is buggy I think -- gaps.largest_gap returns None if there is no gap.